host/app/components/go.gts

```diff
@@ -88,23 +93,30 @@ export default class Go extends Component<Signature> {
         return undefined;
       }
       if (isCardDocument(maybeCard)) {
+        let url = this.args.openFile?.url.replace(/\.json$/, '');
+        taskFor(this.loadData).perform(url);
         return maybeCard;
       }
     }
     return undefined;
   }
 
-  get cardArgs(): ExistingCardArgs {
-    if (this.args.openFile?.state !== 'ready') {
-      throw new Error('No file has been opened yet');
-    }
-    if (!this.openFileCardJSON) {
-      throw new Error('Card JSON is not currently available');
+  @restartableTask private async loadData(url: string | undefined): Promise<void> {
```

Comment:
@burieberry i thought we discussed creating a service for this or is that coming in another PR?